tests/testFWHM.py

```diff
@@ -86,71 +84,59 @@ class characterizing the detector corresponding to this image
-                                     int((iy-y_center)/slope + x_center) < im.shape[1]])
+        ra_p1, dec_p1 = raDecFromPixelCoords(x_center+1, y_center, detector.getName(),
+                                             camera=camera, obs_metadata=obs,
+                                             epoch=epoch)
 
-                xPixList = np.array([int((iy-y_center)/slope + x_center) for iy in yPixList])
+        pixel_scale = angularSeparation(ra_max, dec_max, ra_p1, dec_p1)
+        pixel_scale *= 3600.0  # convert to arcsec
 
-            chipNameList = [detector.getName()]*len(xPixList)
-            raList, decList = _raDecFromPixelCoords(xPixList, yPixList, chipNameList,
-                                                    camera=camera, obs_metadata=obs, epoch=epoch)
+        chisq_best = None
+        fwhm_best = None
 
-            distanceList = arcsecFromRadians(haversine(raList, decList, raMax, decMax))
+        total_flux = im_flat.sum()
 
-            fluxList = np.array([im[iy][ix] for ix, iy in zip(xPixList, yPixList)])
+        for fwhm_test in np.arange(0.1*fwhm, 3.0*fwhm, 0.1*fwhm):
+            alpha = fwhm_test/2.3835
 
-            distanceToLeft = None
-            distanceToRight = None
+            dd = np.power(x_arr-x_center,2).astype(float) + np.power(y_arr-y_center, 2).astype(float)
+            dd *= np.power(pixel_scale, 2)
 
-            for ix in range(1, len(xPixList)):
-                if fluxList[ix] < half_flux and fluxList[ix+1] >= half_flux:
-                    break
+            sigma = alpha
+            g1 = np.exp(-0.5*dd/(sigma*sigma))/(sigma*sigma*2.0*np.pi)
 
-            newOrigin = ix+1
+            sigma = 2.0*alpha
+            g2 = np.exp(-0.5*dd/(sigma*sigma))/(sigma*sigma*2.0*np.pi)
 
-            mm = (distanceList[ix]-distanceList[ix+1])/(fluxList[ix]-fluxList[ix+1])
-            bb = distanceList[ix] - mm * fluxList[ix]
-            distanceToLeft = mm*half_flux + bb
+            model = 0.909*(g1 + 0.1*g2)*pixel_scale*pixel_scale
+            norm = model.sum()
+            model *= (total_flux/norm)
+            chisq = np.power((im_flat-model), 2).sum()
 
-            for ix in range(newOrigin, len(xPixList)-1):
-                if fluxList[ix] >= half_flux and fluxList[ix+1] < half_flux:
-                    break
+            if chisq_best is None or chisq<chisq_best:
+                chisq_best = chisq
+                fwhm_best = fwhm_test
 
-            mm = (distanceList[ix]-distanceList[ix+1])/(fluxList[ix]-fluxList[ix+1])
-            bb = distanceList[ix] - mm * fluxList[ix]
-            distanceToRight = mm*half_flux + bb
-
-            msg = "measured fwhm %e; expected fwhm %e; maxFlux %e; orientation %e pi\n" % \
-                  (distanceToLeft+distanceToRight, fwhm, maxFlux, theta/np.pi)
-
-            self.assertLess(np.abs(distanceToLeft+distanceToRight-fwhm), 0.1*fwhm, msg=msg)
+        self.assertLess(np.abs(fwhm-fwhm_best), 0.1*fwhm)
```

Comment:
Are we sure that, given these images are driven with photon shooting, the test will ever be able to detect the difference between PSFs with and without pixel convolution?  We can either test in the well-sampled regime, in which case 1/12 of a pixel is inconsequential, or we can test in the poorly-sampled regime, in which case the stochasticity of photon shooting will overwhelm 1/12 of the pixel.  That, at least, was my experience when writing the test as-is.  The test was good enough to tell that the PSF was "about right", but it didn't really matter whether the 1/12 was subtracted, or not.  I thought that was due to the nature of photon shooting.  It could mean I was doing something wrong.